### PR TITLE
Unity android ddl

### PR DESCRIPTION
**Pull request by TomBarzilay1**
in this PR we have added deep-linking functionality

### CODE REVIEW

File: unity-sdk/AndroidPlugin/OptimoveUnityPlugin/build.gradle [hunk 1]
```diff
@@ -44,4 +44,7 @@ dependencies {
     androidTestImplementation 'androidx.test.espresso:espresso-core:3.4.0'
     compileOnly 'com.optimove.android:optimove-android:7.+'
     compileOnly files('libs/classes.jar')
+    compileOnly files('libs/firebase-messaging-unity-10.0.1.aar')
```
Comment (vvoicehovics) [inline unity-sdk/AndroidPlugin/OptimoveUnityPlugin/build.gradle]: this not needed anymore + cleanup everywhere

File: unity-sdk/AndroidPlugin/settings.gradle [hunk 2]
```diff
@@ -15,3 +15,4 @@ dependencyResolutionManagement {
 rootProject.name = "unity"
 include ':app'
 include ':OptimoveUnityPlugin'
+include ':UnityActivity'
```
Comment (vvoicehovics) [inline unity-sdk/AndroidPlugin/settings.gradle]: here is a SO question (which mb you saw) https://stackoverflow.com/questions/61865739/cannot-find-the-unityplayeractivity-class-inside-com-unity3d-player

so, this is prob fine. They all suggest copying unity activity one way or another

Comment (vvoicehovics) [reply unity-sdk/AndroidPlugin/settings.gradle]: i can come up with another way via gradle project manipulation (same way as we add google-services.json)
1. copy our overriding activity into streaming assets
2. before gradle is building copy into UnityPlugin module

but perhaps even more hacky / not 'official'

Comment (vvoicehovics) [reply unity-sdk/AndroidPlugin/settings.gradle]: If there is another plugin doing same (extending UnityActivity), consumers could somehow merge all these extensions, so, messy, but possible

File: ExampleApp/Assets/Plugins/Android/AndroidManifest.xml [hunk 3]
```diff
@@ -1,11 +1,18 @@
...
         <intent-filter>
           <action android:name="android.intent.action.MAIN" />
           <category android:name="android.intent.category.LAUNCHER" />
         </intent-filter>
+            <intent-filter android:label="deepLabel" android:autoVerify="true">
+    <action android:name="android.intent.action.VIEW" />
+    <category android:name="android.intent.category.DEFAULT" />
+    <category android:name="android.intent.category.BROWSABLE" />
+    <!-- Specify which URIs should be matched. Note, domain should be .lnk.click -->
+    <data android:scheme="https" android:host="subdomain.lnk.click"/>
```
Comment (vvoicehovics) [inline ExampleApp/Assets/Plugins/Android/AndroidManifest.xml]: better `unity-example-optimove.lnk.click`

File: unity-sdk/AndroidPlugin/OptimoveUnityPlugin/build.gradle (hunk 1 above)
Comment (TomBarzilay1) [reply unity-sdk/AndroidPlugin/OptimoveUnityPlugin/build.gradle]: deleted and removed firebaseMessigng.unityPackage from the project

File: unity-sdk/AndroidPlugin/settings.gradle (hunk 2 above)
Comment (TomBarzilay1) [reply unity-sdk/AndroidPlugin/settings.gradle]: I will add to the docs a section that explains why we have added UnityActivity.aar

File: ExampleApp/Assets/Plugins/Android/AndroidManifest.xml (hunk 3 above)
Comment (TomBarzilay1) [reply ExampleApp/Assets/Plugins/Android/AndroidManifest.xml]: 👍

**Pull request merged**